evm/src/executor/backend/mod.rs

```diff
@@ -848,7 +858,7 @@ impl DatabaseExt for Backend {
                 .map(|addr| self.is_failed_test_contract_state(addr, current_state))
                 .unwrap_or_default()
             {
-                self.inner.has_snapshot_failure = true;
+                self.inner.has_snapshot_failure.fetch_or(true, Ordering::Relaxed);
```

Comment:
this is just store, right?
```suggestion
                self.inner.has_snapshot_failure.store(true, Ordering::SeqCst);
```

Reply:
oh yeah should be, good catch

Reply:
done in https://github.com/foundry-rs/foundry/pull/4974/commits/7fb915a36687bc54e9a32b4637fd2b194762d432